dump/src/reader/v4/mod.rs

```diff
@@ -147,16 +150,45 @@ pub struct V4IndexReader {
 }
 
 impl V4IndexReader {
-    pub fn new(name: String, path: &Path) -> Result<Self> {
+    pub fn new(name: String, path: &Path, tasks: BufReader<File>) -> Result<Self> {
         let meta = File::open(path.join("meta.json"))?;
         let meta: DumpMeta = serde_json::from_reader(meta)?;
 
+        let mut created_at = None;
+        let mut updated_at = None;
+
+        for line in tasks.lines() {
+            let task: Task = serde_json::from_str(&line?)?;
+
+            if task.index_uid.to_string() == name {
+                if updated_at.is_none() {
+                    updated_at = match task.events.last() {
+                        Some(TaskEvent::Created(ts)) => Some(*ts),
+                        _ => None,
+                    };
+                }
+
+                if created_at.is_none() {
+                    created_at = match task.content {
+                        TaskContent::IndexCreation { primary_key } => match task.events.first() {
+                            Some(TaskEvent::Created(ts)) => Some(*ts),
+                            _ => None,
+                        },
+                        _ => None,
+                    };
+                }
```

Comment:
hey @funilrys sorry, when I wrote my first review I totally forgot something about meilisearch.
And index can be created by multiple events:
- An indexCreation
- A documentAddition
- A settingsAddition

Thus this simple if actually doesn't work and finding the creation date is harder than I thought.
BUT, when we wrote this dump format back in the past we already thought about it and created a special struct to retrieve the `creation_task_id`.
Here is a print of the `index_uuids/data.jsonl` file:
```json
{
  "uid": "dnd_spells",
  "index_meta": {
    "uuid": "684b4662-e3c3-4440-8d57-07ee42b0b901",
    "creation_task_id": 6
  }
```

And in the code of the dumpV4 I'm already deserializing this structure:
```rust
#[derive(Deserialize, Debug)]
#[cfg_attr(test, derive(serde::Serialize))]
pub struct IndexUuid {
    pub uid: String,
    pub index_meta: IndexMeta,
}

#[derive(Deserialize, Debug)]
#[cfg_attr(test, derive(serde::Serialize))]
pub struct IndexMeta {
    pub uuid: Uuid,
    pub creation_task_id: usize,
}
```


Thus the idea for
1. Stopping the loop
2. Setting the creation date
Is to simply stop when we reach the `creation_task_id` of the index match the `task_id` of the task.

```suggestion
                if task.id == self.creation_task_id { // I have no idea of what fields are accessible sorry
                    created_at = task.processed_at(); // we'll probably need to write a function that extract the date whatever is the variant
                    break; // we don't need to check anything else and can stop right away
                }
```

Reply:
@irevoire Sorry for the waiting time... Last week was a long one! ... 

I can't find any `TaskContent::SettingsAddition` ... Do you mean `TaskContent::SetingsUpdate` ?

Reply:
By the way, is is a reason behind the incompatibility of `task.id` and `IndexMeta.creation_task_id` ?

One is `u32` and the later `usize` ... That lost me ...

Reply:
@irevoire in your suggestion what would `processed_at` be? Right now I will only shift my match to that method ... I don't really understand what you mean with "whatever is the variant" ... 

Right now - from my understanding - the only way to read the creation or update time is to go through the list of events - as I implemented ...

Reply:
> Sorry for the waiting time...

No worries! I’m already super happy to see someone tackling this hard problem and getting into the codebase!

> Do you mean TaskContent::SettingsUpdate ?

Yes sorry! 🙏 

> By the way, is is a reason behind the incompatibility of task.id and IndexMeta.creation_task_id ?

The reason is because the way I implemented the dumps was by taking the exact code that generated these dumps at the time. And it looks like, at the time of the dumpv4 the `creation_task_id` was a `usize` for no reason, thus we need to convert it as `u32` or the other way around before checking for equality 😔 

> in your suggestion what would processed_at be?

It should be a new method implemented on the `Task` type in the `v4::task` module that returns, at which time the task has been processed.
Being processed means, the task was either a success or a failure. (little note on this: meilisearch isn’t entirely ACID, and when adding documents, an index will be automatically created, but then the document addition can fail; thus, an index can be created by a failed task).

I imagine something like that would be enough:
```rust
impl Task {
    /// Return at which time the task finished processing.
    pub fn processed_at(&self) -> Option<OffsetDateTime> {
        match self.events.last() {
            Some(TaskEvent::Succeded { timestamp, .. } | TaskEvent::Failed { timestamp, .. }) => {
                Some(timestamp.clone())
            }
            _ => None,
        }
    }
}
```

-------

> the only way to read the creation or update time is to go through the list of events - as I implemented ...

Well, yes and no. We still need to go through the list of events, as you said, thus, the code shouldn’t change a lot.
But we need to use the stored `created_at` value to get the time of the task that created the index. And for the last_update part I think your code is working 👍